src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -16,28 +16,51 @@
  */
 
 #include <app/clusters/bindings/BindingManager.h>
+#include <app/clusters/bindings/bindings.h>
 #include <app/util/binding-table.h>
 #include <credentials/FabricTable.h>
+#include <lib/support/CHIPMem.h>
 
 namespace {
 
 class BindingFabricTableDelegate : public chip::FabricTableDelegate
 {
     void OnFabricDeletedFromStorage(chip::CompressedFabricId compressedFabricId, chip::FabricIndex fabricIndex)
     {
-        auto iter = chip::BindingTable::GetInstance().begin();
-        while (iter != chip::BindingTable::GetInstance().end())
+        chip::BindingTable & bindingTable = chip::BindingTable::GetInstance();
+        auto iter                         = bindingTable.begin();
+        while (iter != bindingTable.end())
         {
             if (iter->fabricIndex == fabricIndex)
             {
-                iter = chip::BindingTable::GetInstance().RemoveAt(iter);
+                iter = bindingTable.RemoveAt(iter);
             }
             else
             {
                 ++iter;
             }
         }
         chip::BindingManager::GetInstance().FabricRemoved(compressedFabricId, fabricIndex);
+
+        EmberBindingTableEntry * table = static_cast<EmberBindingTableEntry *>(
+            chip::Platform::MemoryAlloc(bindingTable.Size() * sizeof(EmberBindingTableEntry)));
+        uint8_t idx = 0;
+        if (table == nullptr)
+        {
+            return;
+        }
+        for (const EmberBindingTableEntry & entry : chip::BindingTable::GetInstance())
+        {
+            table[idx++] = entry;
+        }
+        CHIP_ERROR error = chip::Server::GetInstance().GetPersistentStorage().SyncSetKeyValue(
+            chip::kBindingStoargeKey, table, static_cast<uint16_t>(bindingTable.Size() * sizeof(EmberBindingTableEntry)));
```

Comment:
This approach to storage means we can never change the layout of the binding table entries.  That seems pretty fragile.